File: Making_Anagrams/making_anagrams.py
#Ryan Barnard
#Making Anagrams

#This is my python solution to the Making Anagrams problem from hackerrank.com under
#the string manipulation section of the Interview Preparation Kit at:
#https://www.hackerrank.com/challenges/ctci-making-anagrams/problem

def makeAnagram(a, b):
    removals = 0
    a_chars = [0]*26
    b_chars = [0]*26
    for letter in a:
        a_chars[ord(letter)-97] += 1
    for letter in b:
        b_chars[ord(letter)-97] += 1
    for i in range(26):
        removals += abs(a_chars[i]-b_chars[i])
    return removals

def main():
    a = "fcrxzwscanmligyxyvym"
    b = "jxwtrhvujlmrpdoqbisbwhmgpmeoke"
    if makeAnagram(a,b) == 30:
        print("Success")
    else:
        print("Failure")

File: Making_Anagrams/making_anagrams.cpp
//Ryan Barnard
//Making Anagrams

//This is my C++ solution to the Making Anagrams problem from hackerrank.com under
//the string manipulation section of the Interview Preparation Kit at:
//https://www.hackerrank.com/challenges/ctci-making-anagrams/problem

#include <string>
#include <iostream>

int makeAnagram(std::string a, std::string b){
    int removals = 0;
    int arr_a[26] = {0};
    int arr_b[26] = {0};
    for(int i = 0; i < a.size(); i++){
        arr_a[a[i]-97]++;
    }
    for(int j = 0; j < b.size(); j++){
        arr_b[b[j]-97]++;
    }
    for(int k = 0; k < 26; k++){
        removals += abs(arr_a[k] - arr_b[k]);
    }
    return removals;
}

int main(){
    std::string a = "fcrxzwscanmligyxyvym";
    std::string b = "jxwtrhvujlmrpdoqbisbwhmgpmeoke";
    if(makeAnagram(a, b) == 30){
        std::cout << "Success" << std::endl;
    }   
    else{
        std::cout << "Failure" << std::endl;
    }
    return 0;
}